docs(storybook): docs cleanup (#6495)
adobe/spectrum-web-components: 2nd-gen/packages/swc/.storybook/learn-about-swc/gen1-vs-gen2.mdx

## Key differences

| Area               | SWC Gen1                                                                                             | SWC Gen2                                                                                  |
| ------------------ | ---------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- |
| Package name       | `@spectrum-web-components`                                                                           | `@adobe/spectrum-wc`                                                                      |
| Documentation site | [opensource.adobe.com/spectrum-web-components](https://opensource.adobe.com/spectrum-web-components) | [spectrum-web-components.adobe.com](https://spectrum-web-components.adobe.com)            |
| Design             | Includes Spectrum, Express, and S2 Foundations themes                                                | Spectrum 2 only                                                                           |

> | Architecture       | Earlier component patterns; less consistent                                                          | More consistent structure and APIs across the library                                     |

Reviewer: ```suggestion
| Architecture       | Earlier component patterns with less consistency                                                          | Increased consistency of structure and APIs throughout the library                                     |
```

> | Styling            | Components are styled via `<sp-theme>`                                                               | Components styled via CSS classes at application root (e.g., `swc-theme swc-theme--dark`) |

Reviewer: ```suggestion
| Styling            | Components are styled via `<sp-theme>`                                                               | Components are styled via CSS classes at the application root (e.g., `swc-theme swc-theme--dark`) |
```

| Customization      | Done via `--mod` CSS custom properties                                                               | Done via `--swc` custom properties and/or global element styles                           |
| Accessibility      | Basic support, several known bugs                                                                    | Improved accessibility coverage, bugs fixed                                               |
| Performance        | More interdependencies, larger bundles                                                               | Smaller bundles, higher reliance on browser standards, better runtime performance         |
| Long-term support  | No new components or features, minimal bug fixes                                                     | New features, components, and fixes                                                       |

## What this means for teams

> ### If you are starting a new project

Reviewer: ```suggestion
### If you're starting a new project
```

Use Gen2 where available. It matches the most recent Spectrum guidelines, is built on a new architecture designed to scale into the future, and will receive long-term support.

### If your project currently uses Gen1 SWC

> You should begin to plan how you'll adopt Gen2 components as they become available. Each component has a migration guide which includes more details on the specific changes. Migrating on a per-component basis allows teams to update gradually, and the Gen1 "spectrum-two" theme allows for visual consistency between both generations.

Reviewer: ```suggestion
You should begin to plan how you'll adopt Gen2 components as they become available. Each component has a migration guide, which includes details about specific changes. Migrating on a per-component basis allows teams to update gradually, and the Gen1 `spectrum-two` theme provides visual consistency between both generations.
```

> For a step-by-step walkthrough of coexisting Gen1 `<sp-*>` and Gen2 `<swc-*>` elements, swapping themes and fonts, mapping `--mod-*` overrides to `--swc-*`, and decommissioning Gen1, see [Migrate from Gen1](/docs/resources-migrate-from-gen1--docs).

Reviewer: ```suggestion
For a step-by-step walkthrough of how to manage coexisting Gen1 `<sp-*>` and Gen2 `<swc-*>` elements, swap themes and fonts, map `--mod-*` overrides to `--swc-*`, and decommission Gen1, see [Migrate from Gen1](/docs/resources-migrate-from-gen1--docs).
```

## Learn more

> You can find specifics about each component in its documentation, including updated APIs, usage examples, patterns, and migration guides.

Reviewer: ```suggestion
You can find specifics about each component in its documentation, which includes updated APIs, usage examples, patterns, and migration guides.
```

> For more details on the changes to our styling and customization, our [customization guides](/docs/guides-customization-getting-started--docs) are a great place to start.

Reviewer: ```suggestion
To learn more about changes to styling and customization in Gen2, refer to our [customization guides](/docs/guides-customization-getting-started--docs).
```

> You can find more details about our bug policy on our [Support and Compatibility page](/docs/resources-support-and-compatibility--docs).

Reviewer: ```suggestion
More details about our bug policy can be found on our [Support and Compatibility page](/docs/resources-support-and-compatibility--docs).
```
